### ISSUE TITLE
Signals prior to spawning point / jump to station / are not set properly

### ISSUE THREAD

**Issue by custodian** [Libre-TrainSim/Libre-TrainSim#534]
### Summary

Signals prior to spawning point are not set properly, i.e. remain green

### How to reproduce

i.
1. Launch any scenario with 3+ stations
2a. Jump to station.
2b. Find NPC train just spawned.
3. Check signals state before the spawn point

ii. 
1. Launch any scenario and spawn at the middle of the track
2. Check signals state before the spawn point.

### Observed behavior

Signals are green:
Spawn at the middle of the track:
![image](https://github.com/Libre-TrainSim/Libre-TrainSim/assets/935667/5daae555-d15c-4bbc-be20-9484d3fc9581)

NPC train:
![image](https://github.com/Libre-TrainSim/Libre-TrainSim/assets/935667/6570b241-8c6d-4732-8e66-f8932e0a3f9e)

### Expected behavior

Signals are Yellow/Red
![image](https://github.com/Libre-TrainSim/Libre-TrainSim/assets/935667/2d60e1fa-eab7-4055-8566-335e8fdbe5ba)

### Version

- Game Version: master 3ffba0dc29d78ac237606524f0a782a3ef360ed3

### Additional information

Linked to #315

**Comment by DasCapschen** [comment]
Yeah, unfortunately the block system is very basic right now. Right now, signals switch themselves (and the signal before them) the moment a train drives over the signal. The "jump to station" feature just ignores this.  

I want to completely refactor the signal system, but that's only going to happen in 0.10 unfortunately
